src/libcore/num/mod.rs

```diff
@@ -1058,7 +1058,7 @@ $EndFeature, "
             #[inline]
             pub fn saturating_mul(self, rhs: Self) -> Self {
                 self.checked_mul(rhs).unwrap_or_else(|| {
-                    if (self < 0 && rhs < 0) || (self > 0 && rhs > 0) {
+                    if (self < 0) == (rhs < 0) {
```

Comment:
This is actually also more readable than before IMO, nice :)

Reply:
And in this case I think constness just has to wait :)